internal/processors/token_transfer.go

```diff
@@ -0,0 +1,350 @@
+	amount := transfer.GetAmount()
+	asset := transfer.GetAsset()
+
+	switch *operationType {
+	case xdr.OperationTypeCreateClaimableBalance:
+		// When creating a claimable balance, record debit from creator with CB ID
+		change := p.createClaimableBalanceChange(types.StateChangeCategoryDebit, transfer.GetFrom(), transfer.GetTo(), amount, asset, contractAddress, builder)
+		return []types.StateChange{change}, nil
+
+	case xdr.OperationTypeClaimClaimableBalance:
+		// When claiming a claimable balance, record credit to claimer with CB ID
+		change := p.createClaimableBalanceChange(types.StateChangeCategoryCredit, transfer.GetTo(), transfer.GetFrom(), amount, asset, contractAddress, builder)
+		return []types.StateChange{change}, nil
+
+	case xdr.OperationTypeLiquidityPoolDeposit:
+		// When depositing to LP, record debit from depositor with LP ID
+		change := p.createLiquidityPoolChange(types.StateChangeCategoryDebit, transfer.GetFrom(), transfer.GetTo(), amount, asset, contractAddress, builder)
+		return []types.StateChange{change}, nil
+
+	case xdr.OperationTypeLiquidityPoolWithdraw:
+		// When withdrawing from LP, record credit to withdrawer with LP ID
+		change := p.createLiquidityPoolChange(types.StateChangeCategoryCredit, transfer.GetTo(), transfer.GetFrom(), amount, asset, contractAddress, builder)
+		return []types.StateChange{change}, nil
+
+	case xdr.OperationTypeSetTrustLineFlags, xdr.OperationTypeAllowTrust:
+		// Skip events generated by these operations since they involve only an LP and Claimable Balance ID
+		return nil, nil
+
+	default:
+		// Handle regular transfers, path payments, and other operations
+		return p.handleDefaultTransfer(transfer, contractAddress, builder)
+	}
+}
+
+// handleDefaultTransfer handles normal transfers and liquidity pool interactions.
+// For LP interactions, creates single state change with LP ID. For regular transfers, creates debit/credit pair.
+func (p *TokenTransferProcessor) handleDefaultTransfer(transfer *ttp.Transfer, contractAddress string, builder *StateChangeBuilder) ([]types.StateChange, error) {
+	from := transfer.GetFrom()
+	to := transfer.GetTo()
+	amount := transfer.GetAmount()
+	asset := transfer.GetAsset()
+
+	fromIsLP := IsLiquidityPool(from)
+	toIsLP := IsLiquidityPool(to)
+
+	// Handle liquidity pool interactions (from path payments, manual LP operations)
+	if fromIsLP || toIsLP {
+		if fromIsLP {
+			// LP is sending tokens to account (e.g., path payment buying from LP)
+			change := p.createLiquidityPoolChange(types.StateChangeCategoryCredit, to, from, amount, asset, contractAddress, builder)
+			return []types.StateChange{change}, nil
+		}
+		// LP is receiving tokens from account (e.g., path payment selling to LP)
+		change := p.createLiquidityPoolChange(types.StateChangeCategoryDebit, from, to, amount, asset, contractAddress, builder)
+		return []types.StateChange{change}, nil
+	}
```

Comment:
Considering it's not possible for both `fromIsLP` and  `toIsLP`  to be true, can we simplify things a bit?

```suggestion
	if fromIsLP {
		// LP is sending tokens to account (e.g., path payment buying from LP)
		change := p.createLiquidityPoolChange(types.StateChangeCategoryCredit, to, from, amount, asset, contractAddress, builder)
		return []types.StateChange{change}, nil
	} else if toIsLP {
		// LP is receiving tokens from account (e.g., path payment selling to LP)
		change := p.createLiquidityPoolChange(types.StateChangeCategoryDebit, from, to, amount, asset, contractAddress, builder)
		return []types.StateChange{change}, nil
	}
```